### PR TITLE
Input/Output Size should be -1 if shape has negative elements

### PR DESCRIPTION
**Pull request by apivovarov**
Input/Output Size should be -1 if shape has negative elements

### CODE REVIEW

**Review by trevor-m — APPROVED** [review]
Nice, thanks @apivovarov !

**Comment by cnv1989** [comment]
Let me check with smith to see if they are fine with this.

File: src/dlr_common.cc [hunk 1]
```diff
@@ -195,3 +195,7 @@ DLDeviceType dlr::GetDeviceTypeFromMetadata(const std::vector<std::string>& mode
     throw dmlc::Error("TargetDeviceType was not found in metadata!");
   } 
 }
+
+bool dlr::HasNegative(const int64_t* arr, const size_t size) {
```
Comment (mycpuorg) [inline src/dlr_common.cc:199]: nit:
```
dlr::shapeHasNegative()
```

**Review by cnv1989 — APPROVED** [review]

File: src/dlr_relayvm.cc [hunk 2]
```diff
@@ -278,7 +279,11 @@ void RelayVMModel::GetOutputSizeDim(int index, int64_t* size, int* dim) {
   *size = 1;
   if (index < outputs_.size()) {
     auto arr = outputs_[index];
-    *size = std::accumulate(arr->shape, arr->shape + arr->ndim, 1, std::multiplies<int64_t>());
+    if (dlr::HasNegative(arr -> shape, arr -> ndim)) {
+      *size = -1;
```
Comment (mycpuorg) [inline src/dlr_relayvm.cc:283]: This could be an error code like
```
BAD_SIZE_ERR
```

File: src/dlr_tvm.cc [hunk 3]
```diff
@@ -204,6 +205,10 @@ void TVMModel::GetOutputSizeDim(int index, int64_t* size, int* dim) {
   *size = 1;
   const DLTensor* tensor = outputs_[index];
   for (int i = 0; i < tensor->ndim; ++i) {
+    if (tensor->shape[i] < 0) {
```
Comment (mycpuorg) [inline src/dlr_tvm.cc:208]: Overall, there is value in this change. Can we add unit tests for this change and add this to the documentation of the `GetOutputSizeDim()` and `GetInputDim()`

File: src/dlr_relayvm.cc (hunk 2 above)
Comment (trevor-m) [reply src/dlr_relayvm.cc:283]: -1 is a valid size for RelayVM models, this PR is just so GetOutputSizeDim returns a consistent value for an unknown size

Comment (mycpuorg) [reply src/dlr_relayvm.cc:283]: I misunderstood your response:
All I was saying is:
```c++
constexpr int UNKNOWN_SIZE = -1;
```

**Pull request merged**